packages/gestalt/src/SlimBanner.js

```diff
@@ -216,16 +218,21 @@ export default function SlimBanner({
           <Flex.Item flex="none">
             <Flex alignItems="center" gap={{ row: 4, column: 0 }}>
               {primaryAction && (
-                <Flex.Item flex="none">
+                <Box display="none" mdDisplay="flex" flex="none">
                   <PrimaryAction {...primaryAction} />
-                </Flex.Item>
+                </Box>
               )}
 
               {dismissButton && <DismissButton {...dismissButton} />}
             </Flex>
           </Flex.Item>
         )}
       </Flex>
+      {shouldShowButtons && primaryAction && (
```

Comment:
isn't "shouldShowButtons" redundant?

isn't just `!isBare` what we need? we double check for && primaryAction 
&& dismissButton before each component

const shouldShowButtons = !isBare && (primaryAction || dismissButton);

Reply:
switched it to !isBare && primaryAction since flow needs the primaryAction check still